File: javascript/fibonacci.js
function fibonacci(num) {
  let array = []
  let element1
  let element2
  for (let i = 0; i <= num; i++) {
    if (array[array.length -2]) {
      element1 = array[array.length -2]
    } else {
      element1 = 0
    }
    if (array[array.length -1]) {
      element2 = array[array.length -1]
    } else {
      element2 = 0
    }
    sum = element1 + element2
    if (i === 1) {
      array.push(1)
    } else {
      array.push(sum)    
    }
  }
  return array[array.length-1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  array = []
  repeat = num + 1
  repeat.times do |i|
    array[-1] ? element1 = array[-1] : element1 = 0
    array[-2] ? element2 = array[-2] : element2 = 0
    sum = element1 + element2
    i == 1 ? array << 1 : array << sum
  end
  array[-1]
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
